File: 0389-find-the-difference/0389-find-the-difference.kt
class Solution {
    fun findTheDifference(s: String, t: String): Char {
        val k = s + t
        var n = 0
        for (i in 0..(k.length-1)) {
            n = n xor k[i].toInt()
        }
        return n.toChar()
    }
}

File: 0389-find-the-difference/0389-find-the-difference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        map_ = {}
        for c in s:
            map_[c] = map_.get(c, 0) + 1
        for c in t:
            if c not in map_ or map_[c] == 0:
                return c
            map_[c] -= 1